src/backend/joanie/core/api/client/__init__.py

```diff
@@ -541,6 +542,50 @@ def withdraw(self, request, pk=None):  # pylint: disable=no-self-use, invalid-na
 
         return Response(status=HTTPStatus.NO_CONTENT)
 
+    @action(detail=True, methods=["POST"])
+    def pay_failed_installment_payment(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name, unused-argument
+        """
+        Pay a failed installment that was scheduled for a given order.
+        """
+        order = self.get_object()
+        if order.state != enums.ORDER_STATE_FAILED_PAYMENT:
+            return Response(
+                {
+                    "detail": "The order is not in failed payment state for last payment."
+                },
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+        payment_backend = get_payment_backend()
+        installment = order.get_first_installment_refused()
+
+        card_id = request.data.get("card_id")
+        if not card_id:
+            payment_infos = payment_backend.create_payment(
+                order=order,
+                billing_address=order.main_invoice.recipient_address,
+                installment=installment,
+            )
+            return Response(payment_infos, status=HTTPStatus.OK)
+
+        credit_card = CreditCard.objects.get_card_for_owner(
+            token=card_id, user_id=order.owner.id
```

Comment:
No the `token` is never expose to the front application. You will receive the `id` of the `CreditCard` model here. Not the `token`

Reply:
Sorry I got confused the other day between the 'card_id' and the 'card_token_id'. Correcting it right away.